pkg/operator/ceph/object/spec.go

```diff
@@ -179,6 +180,28 @@ func (c *clusterConfig) makeRGWPodSpec(rgwConfig *rgwConfig) (v1.PodTemplateSpec
 			podSpec.Volumes = append(podSpec.Volumes, vaultFileVol, tmpvolume)
 			podSpec.InitContainers = append(podSpec.InitContainers,
 				c.vaultTokenInitContainer(rgwConfig))
+		} else if c.store.Spec.Security.KeyManagementService.IsK8sAuthEnabled() {
+			vaultFileVol, _ := kms.VaultVolumeAndMount(c.store.Spec.Security.KeyManagementService.ConnectionDetails,
+				"")
+			tmpvolume := v1.Volume{
+				Name: rgwVaultVolumeName,
+				VolumeSource: v1.VolumeSource{
+					EmptyDir: &v1.EmptyDirVolumeSource{},
+				},
+			}
+
+			podSpec.Volumes = append(podSpec.Volumes, vaultFileVol, tmpvolume)
+			vaultAgentVol := v1.Volume{Name: "vault-agent-config",
+				VolumeSource: v1.VolumeSource{
+					ConfigMap: &v1.ConfigMapVolumeSource{
+						LocalObjectReference: v1.LocalObjectReference{Name: "vault-agent-cm"},
```

Comment:
```yaml
apiVersion: v1
data:
  vault-agent-config.hcl: |
    # Comment this out if running as sidecar instead of initContainer
    # exit_after_auth = true

    pid_file = "/home/vault/pidfile"

    auto_auth {
        method "kubernetes" {
            mount_path = "auth/kubernetes"
            config = { 
                role = "rook-ceph"
            }
        }
    }   
    cache {
      use_auto_auth_token = true
    }   
    listener "tcp" {
      address = "127.0.0.1:8100"
      tls_disable = "true"
    }   
    vault {
      address = "https://vault.default:8200"
      tls_key_file = <filename>
      tls_cert_file = <filename>
      tls_client_ca_file = <filename>
    }   

kind: ConfigMap
metadata:
  name: vault-agent-cm
  namespace: rook-ceph
```
Above is the cm for the vault agent config, do we ask users input the cm or can we create it for them. 
If we create cm for them, then Rook can only support vault agent via service account

Reply:
If we think the users might need to modify and input their own values then let's ask otherwise let the operator handle all of that.

Reply:
I guess then it is okay for the Operator to handle it. The input will be given in `ObjectStore.Spec.Security` similar to OSD Encryption instead passing to RGW server, we use that for creating vault agent sidecar container